checker/src/main/java/alluxio/checker/HiveIntegrationChecker.java

```
@@ -0,0 +1,219 @@
+    }
+  }
+
+  /**
+   * Supports the second integration way: use Alluxio as the default filesystem.
+   *
+   * @param con the Hive connection
+   * @param tableName the name of the test table
+   */
+  private void useAsUnderFS(Connection con, String tableName) throws Exception {
+    String sql = "CREATE TABLE " + tableName + " (ROW1 STRING, ROW2 STRING) "
+        + "ROW FORMAT DELIMITED FIELDS TERMINATED BY '|' "
+        + "STORED AS TEXTFILE";
+    try (PreparedStatement createTablePS = con.prepareStatement(sql)) {
+      createTablePS.execute();
+    }
+
+    sql = "INSERT INTO " + tableName + " VALUES ('Hive', 'Test')";
+    try (PreparedStatement loadTablePS = con.prepareStatement(sql)) {
+      loadTablePS.executeUpdate();
+    }
+  }
+
+  /**
+   * Translates the exception message to user-facing message.
+   *
+   * @param exception the thrown exception when running Hive queries
+   * @param reportWriter save user-facing messages to a generated file
+   */
+  private void printExceptionReport(Exception exception, PrintWriter reportWriter) {
+    String exceptionStr = exception.toString();
+    if (exceptionStr.contains("Class alluxio.hadoop.FileSystem not found")) {
+      reportWriter.println(FAIL_TO_FIND_CLASS_MESSAGE);
+      reportWriter.println(TEST_FAILED_MESSAGE);
+    } else if (exceptionStr.contains("No FileSystem for scheme \"alluxio\"")
+        || exceptionStr.contains("No FileSystem for scheme: alluxio")) {
+      reportWriter.println(FAIL_TO_FIND_FS_MESSAGE);
+      reportWriter.println(TEST_FAILED_MESSAGE);
+    } else {
+      reportWriter.println("Other errors occur, please fix it and "
+          + "rerun the Hive Integration Checker. \n");
+      exception.printStackTrace(reportWriter);
+      reportWriter.println(TEST_FAILED_MESSAGE);
+    }
+  }
+
+  /**
+   * Checks if the input arguments are valid.
+   *
+   * @return true if input is valid, false otherwise
+   */
+  private boolean checkIfInputValid() {
```

typically in our codebase, for `checkXXX` we throw RTE `IllegalArgumentException` rather than return a boolean value